test/lib/samson/secrets/hashicorp_vault_backend_test.rb

```diff
@@ -144,10 +144,6 @@
     it "leaves out vaules from deploy groups that have no vault server so KeyResolver works" do
       backend.read_multi(['production/foo/pod100/bar']).must_equal({})
     end
-
-    it "leaves out vaules from unknown deploy groups" do
-      backend.read_multi(['production/foo/pod1nope/bar']).must_equal({})
-    end
```

Comment:
... we still need a testcase for this ?

Reply:
oops, I thought this was testing the same functionality as https://github.com/zendesk/samson/pull/3709/files/430a0f5d71795fde0bcca0ad610ecd4ce3a3c6c7#diff-2f2df8cde92ee51d143c321a9d28f391R145 as I didn't realize a pod100 deploy group existed for the test 😅 

I've added the test back